.github/workflows/publish.yml

```
@@ -119,6 +119,43 @@ jobs:
           }
           program Microsoft.SourceLink.GitHub
 
+      - name: Update package manifest version
+        shell: pwsh
+        run: |
+            function Program {
+                param(
+                    [string]
+                    $WorkDir,
+                    [string]
+                    $PackageVersion
+                )
+
+                function Update-Manifest-Version {
+                    param(
+                        [string]
+                        $Manifest,
+                        [string]
+                        $Version
+                    )
+
+                    (Get-Content -Raw $Manifest)`
+                        -replace '(?<head>\[assembly:\s*Module\(([^\]]*Version\W*=\W*"))([^"]*)', ('${head}' + $Version)` |
+                        Out-File $Manifest
```

I really dislike the sight of escaping newlines (or any kind of non-printing character).
Also why are you not interpolating `$Version` too?
```suggestion
                    (Get-Content -Raw $Manifest) -replace 
                        '(?<head>\[assembly:\s*Module\(([^\]]*Version\W*=\W*"))([^"]*)', '${head}${Version}' |
                        Out-File $Manifest
```